index.html

```diff
@@ -1321,12 +1288,87 @@
       <p>
         The following example shows how to represent US$55.00.
       </p>
-      <pre class="example">
-{
-  "currency": "USD",
-  "value" : "55.00"
-}
+      <pre class="example js">
+        {
+          "currency": "USD",
+          "value" : "55.00"
+        }
       </pre>
+      <section>
+        <h3>
+          Validity checkers
+        </h3>
+        <p>
+          A <dfn data-cite="INFRA#javascript-string">JavaScript string</dfn> is
+          a <dfn>valid decimal monetary value</dfn> if it consists of the
+          following <dfn data-lt="code point" data-cite="INFRA#code-point">code
+          points</dfn> in the given order: [[!INFRA]]
+        </p>
+        <ol>
+          <li>Optionally, a single U+002D (-), to indicate that the amount is
+          negative.
+          </li>
+          <li>One or more <a>code points</a> in the range U+0030 (0) to U+0039
+          (9).
+          </li>
+          <li>Optionally, a single U+002E (.) followed by one or more <a>code
+          points</a> in the range U+0030 (0) to U+0039 (9).
+          </li>
+        </ol>
+        <div class="note">
+          The following regular expression is an implementation of the above
+          definition.
+          <pre class="hljs">^-?[0-9]+(\.[0-9]+)?$</pre>
+        </div>
+        <p>
+          To <dfn data-lt="check and canonicalize amount">check and
+          canonicalize <var>amount</var> <a>PaymentCurrencyAmount</a></dfn>,
+          run the following steps:
+        </p>
+        <ol data-link-for="PaymentCurrencyAmount">
+          <li>If <var>amount</var>.<a>currencySystem</a> is not
+          "<code>urn:iso:std:iso:4217</code>", terminate this algorithm.
+          </li>
+          <li>Let <var>isValidCurrency</var> be the result of calling
+          <a data-cite=
+          "!ecma-402#sec-iswellformedcurrencycode">IsWellFormedCurrencyCode</a>
+          abstract operation with <var>amount</var>.<a>currency</a> as the
+          argument.
+          </li>
+          <li>If <var>isValidCurrency</var> is false, then throw a
+          <a>RangeError</a> exception and terminate this algorithm, optionally
+          informing the developer that the currency is invalid.
+          </li>
+          <li>Let <var>value</var> be <var>amount</var>.<a>value</a>.
+          </li>
+          <li>If <var>value</var> is not a <a>valid decimal monetary value</a>,
+          throw a <a>TypeError</a>, optionally informing the developer that the
+          currency is invalid.
+          </li>
+          <li>Set <var>amount</var>.<a>currency</a> to the result of
+          <a data-cite="!INFRA#ascii-uppercase">ASCII uppercasing</a>
+          <var>amount</var>.<a>currency</a>.
+          </li>
+        </ol>
+        <p>
+          To <dfn data-lt="check and canonicalize total">check and canonicalize
+          <var>total</var> <a>PaymentCurrencyAmount</a></dfn>, run the
```

Comment:
This should be `<dfn>check and canonicalize total</dfn> given a <a>PaymentCurrencyAmount</a> <var>total</var>`, I think.

(Same for the "amount" version)